### ISSUE TITLE
Jetpack Pricing Page: FAQ items have no focus state

### ISSUE THREAD

**Issue by monsieur-z** [Automattic/wp-calypso#74166]
### Quick summary

In the [Jetpack Pricing page](https://cloud.jetpack.com/pricing), items of the FAQ section don't have any visible focus state. This makes navigating this section with a keyboard challenging.

<img width="500" alt="Screenshot 2023-03-07 at 11 46 16 AM" src="https://user-images.githubusercontent.com/1620183/223492650-13956082-9c67-4513-8984-4990ff0aa27c.png">

### Steps to reproduce

1. Visit https://cloud.jetpack.com/pricing
2. Scroll to the _Frequently Asked Questions_ section at the bottom of the page
3. Click on the section title so that the next element in the section will receive focus
4. Click the _Tab_ key several times
5. Notice that you don't **see** anything happening, though FAQ items have focus

Note: you can see what element of the DOM has focus by running `document.activeElement` in the dev's console.

### What you expected to happen

FAQ items should have a visible indicator that they are focused.

### What actually happened

Items have no visible indicator.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable
